common/src/main/java/io/pravega/common/concurrent/DelayedProcessor.java

```diff
@@ -101,19 +104,23 @@ public DelayedProcessor(Function<T, CompletableFuture<Void>> itemProcessor, Dura
 
     //endregion
 
-    //region AutoCloseable implementation
-
-    @Override
-    public void close() {
+    /**
+     * Shuts down the Delay processor. No further items will be processed.
+     * Outstanding items are returned. 
+     * 
+     * @return A list of unprocessed items.
+     */
+    public List<T> shutdown() {
         if (!this.closed.getAndSet(true)) {
-            this.runTask.cancel(true);
             val delayTask = this.currentIterationDelayTask;
             if (delayTask != null) {
                 delayTask.cancel(true);
             }
-            this.queue.clear();
+            Futures.await(runTask);
```

Comment:
Can this be waiting indefinitely in any scenario ? We are not passing any timeout here.

Reply:
This will block until the close is detected on line 169 below. 
The only way it would block forever is if the `runOneIteration` never completes. If this occurs, it's probably the right thing to do, that way the test or shutdown will get stuck on the stuck process.